databricks/koalas/series.py

```diff
@@ -2650,6 +2650,60 @@ def _diff(self, periods, part_cols=()):
         scol = self._scol - F.lag(self._scol, periods).over(window)
         return Series(self._kdf._internal.copy(scol=scol), anchor=self._kdf).rename(self.name)
 
+    # TODO: add frep and axis parameter
+    def shift(self, periods=1, fill_value=None):
+        """
+        Shift DataFrame by desired number of periods.
+
+        .. note:: the current implementation of shift uses Spark's Window without
+            specifying partition specification. This leads to move all data into
+            single partition in single machine and could cause serious
+            performance degradation. Avoid this method against very large dataset.
+
+        Parameters
+        ----------
+        periods : int
+            Number of periods to shift. Can be positive or negative.
+        fill_value : object, optional
+            The scalar value to use for newly introduced missing values.
+            The default depends on the dtype of self. For numeric data, np.nan is used.
+        Returns
+        -------
+        Copy of input DataFrame, shifted.
+        Examples
+        --------
+        >>> df = ks.DataFrame({'Col1': [10, 20, 15, 30, 45],
+        ...                    'Col2': [13, 23, 18, 33, 48],
+        ...                    'Col3': [17, 27, 22, 37, 52]},
+        ...                   columns=['Col1', 'Col2', 'Col3'])
+        >>> df['Col1'].shift(periods=3)
+        0     NaN
+        1     NaN
+        2     NaN
+        3    10.0
+        4    20.0
+        Name: Col1, dtype: float64
+        >>> df['Col1'].shift(periods=3, fill_value=0)
+        0     0
+        1     0
+        2     0
+        3    10
+        4    20
+        Name: Col1, dtype: int64
+        """
+        return self._shift(periods, fill_value)
+
+    def _shift(self, periods, fill_value, part_cols=()):
+        if len(self._internal.index_columns) == 0:
+            raise ValueError("Index must be set.")
+        if not isinstance(periods, int):
+            raise ValueError('periods should be an int')
+        window = Window.partitionBy(*part_cols).orderBy(self._internal.index_scols)\
+            .rowsBetween(-periods, -periods)
+        scol = F.when(F.lag(self._scol, periods).over(window).isNotNull(),
+                      F.lag(self._scol, periods).over(window)).otherwise(fill_value)
```

Comment:
I would write it:

```python
lag_col = F.lag(self._scol, periods).over(window)
scol = F.when(lag_col.isNull(), fill_value).otherwise(lag_col)
```